specutils/fitting/spline_continuum.py

```diff
@@ -0,0 +1,485 @@
+
+    if not isinstance(model, modeling.FittableModel):
+        raise ValueError('The model parameter must be a astropy.modeling.FittableModel object')
+    # TODO: this is waiting on a refactor in modeling.fitting to work
+
+    # if not isinstance(fitter, fitting.Fitter):
+    #     raise ValueError("The model parameter must be an "
+    #                      "astropy.modeling.fitting.Fitter object.")
+
+    # Get input spectrum data
+    x = spectrum.spectral_axis.value
+    y = spectrum.flux.value
+
+    # Set up valid pixels mask. Exclude non-finite values.
+    good = np.isfinite(y)
+
+    # Exclude regions
+    for (excl1, excl2) in exclude_regions:
+        good[np.logical_and(x > excl1, x < excl2)] = False
+
+    # Set up sigma clipping
+    if sigma_lower is None:
+        sigma_lower = sigma
+
+    if sigma_upper is None:
+        sigma_upper = sigma
+
+    # Set the model as the default continuum in cases where the sigma
+    # clipping iterations == 0
+    continuum_model = model
+
+    for i_iter in range(iters):
+        logging.info("Iter {}: Fitting {}/{} pixels".format(
+            i_iter, good.sum(), len(good)))
+
+        # Fit model
+        # TODO: include data uncertainties
+        continuum_model = fitter(model, x[good], y[good])
+
+        # Sigma clip
+        difference = continuum_model(x) - y
+        finite = np.isfinite(difference)
+        sigma_difference = difference / np.std(difference[np.logical_and(good, finite)])
+        good[sigma_difference > sigma_upper] = False
+        good[sigma_difference < -sigma_lower] = False
+
+    if full_output:
+        return continuum_model, good
+
+    return continuum_model
+
+
+def fit_continuum_linetools(spec, edges=None, ax=None, debug=False, kind="QSO", **kwargs):
+    """
+    A direct port of the linetools continuum normalization algorithm by
+    X Prochaska (https://github.com/linetools/linetools/blob/master/linetools/analysis/continuum.py)
+
+    The only changes are switching to Scipy's Akima1D interpolator and
+    changing the relevant syntax.
+    """
+    assert kind in ["QSO"], kind
+
+    if not isinstance(spec, Spectrum1D):
+        raise ValueError('The spectrum parameter must be a Spectrum1D object')
+
+    # To start, we define all the functions here to avoid namespace bloat,
+    # but this can be fixed later. The goal is to have the same algorithm but
+    # with flexible wavelength chunks for other object types
+
+    def make_chunks_qso(wa, redshift, divmult=1, forest_divmult=1,
+                        debug=False):
+        """
+        Generate a series of wavelength chunks for use by
+        prepare_knots, assuming a QSO spectrum.
+        """
+        cond = np.isnan(wa)
+
+        if np.any(cond):
+            warnings.warn('Some wavelengths are NaN, ignoring these pixels.')
+            wa = wa[~cond]
+
+            assert len(wa) > 0
+
+        zp1 = 1 + redshift
+        div = np.rec.fromrecords([(200. , 500. , 25),
```

Comment:
Again, this assumes the spectrum is in wavelength space, which we can't guarantee.